lib/adagrams.rb

```diff
@@ -0,0 +1,127 @@
+require 'csv'
+
+# wave 1
+def stores_letters
+  letters = [ "A", "A", "A", "A", "A", "A", "A", "A", "A", "B", "B", "C", "C", "D", "D", "D", "D", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "F", "F", "G", "G", "G", "H", "H", "I", "I", "I", "I", "I", "I", "I", "I", "I", "J", "K", "L", "L", "L", "L", "M", "M", "N", "N", "N", "N", "N", "N", "O", "O", "O", "O", "O", "O", "O", "O", "P", "P", "Q", "R", "R", "R", "R", "R", "R", "S", "S", "S", "S", "T", "T", "T", "T", "T", "T", "U", "U", "U", "U", "V", "V", "W", "W", "X", "Y", "Y", "Z" ]
+end
+
+def draw_letters
+  letter = stores_letters.shuffle.pop(10)
+  return letter
+end
+
+def welcome
+  puts "Welcome to adagrams. Here are your letters."
+  draw_letters
+end
+
+
+def game_instructions
+  puts "Based on these letters, give us a word."
+  print "Word: "
+end
+
+# def get_user_input
+#   user_input = gets.chomp
+#   return user_input
+# end
+
+# wave 2
+def uses_available_letters?(input, letters_in_hand)
+  input = input.upcase.split("")
+  input.each do |letter|
+    if letters_in_hand.include?(letter)
```

Comment:
There are two problems with this implementation:

1. You will never return `true` for a word with a repeated letter. For example,
    ```ruby
    hand = ["E", "F", "R", "A", "E", "L", "V", "A", "D", "K"]
    uses_available_letters?("ADA", hand)
    # => false
    ```
    The reason why is that Ruby's `Array#delete` method will remove all matching elements from an array, not just the first match.

1. This method is destructive! Because you call `delete` on the input, the hand will be reduced by the letters in the input. For example:
    ```ruby
    hand = draw_letters
    puts hand
    # => ["T", "T", "E", "F", "R", "A", "E", "L", "V", "A"]
    uses_available_letters?('tte', hand)
    puts hand
    # => ["E", "F", "R", "A", "E", "L", "V", "A"]
    ```

    This is bad because it's unexpected behavior. Nothing about "check whether this word is made up of letters in this hand" suggests that the hand will be changed in the process, but that's exactly what happens.

    The way to address this problem would be to make a copy of the hand before removing letters from it.

I will acknowledge that none of the tests checked for either of these behaviors, but that doesn't change the fact that they're bugs.